src/private_benches/CT_README.md

```diff
@@ -0,0 +1,29 @@
+## Constant-Time Benchmarks
+
+This directory contains benchmarks designed to assess the constant-time behavior of `synedrion` code using the Dudect method ([https://eprint.iacr.org/2016/1123.pdf](https://eprint.iacr.org/2016/1123.pdf)). Dudect employs [Welch's t-test](https://en.wikipedia.org/wiki/Welch%27s_t-test) to statistically analyze execution time differences between "normal" and "special" input classes.
+
+### Methodology
+
+1. **Input Generation:** Two input classes are created: "normal" (random) and "special" (potentially triggering timing variations).
+2. **Timing Measurement:** Execution time of the target code is measured for each input.
+3. **Statistical Analysis:** Welch's t-test is applied to compare the execution time distributions of the two classes. The "max t" statistic is computed.
+
+### Interpretation
+
+A small absolute value of "max t" suggests that execution times for the two input classes are similar, indicating potential constant-time behavior.  A threshold of `|max t| >= 5` is often used to flag non-constant-time behavior. However, constant-time analysis can be complex, and a high "max t" value might not always indicate a practically exploitable timing leak.
```

Comment:
Is there any source for the threshold choice?

Reply:
The paper itself uses `4.5` and the `dudect-bencher` crate picks `5` without explicit justification afaict.

I think the choice of `5` as a threshold comes from several directions:

- They do multiple statistical tests (100 I think), discarding different percentiles for each test; when you do this you need stricter threshold than normal.
- It works well empirically. What actually matters here is that non-CT code will make `t` grow with the number of samples, while CT code stays constant. 5 seems to work well in practice. 

I was a bit unsure about just how much commentary to put here. Maybe this stuff is obvious to people that know some statistics? Or maybe not?